spec-tests/src/test/java/org/candlepin/spec/content/ContentVersioningSpecTest.java

```diff
@@ -0,0 +1,314 @@
+
+        assertThat(adminClient.ownerContent().listOwnerContent(owner2.getKey()))
+            .singleElement()
+            .returns(owner2Content.getUuid(), ContentDTO::getUuid);
+    }
+
+    @Test
+    public void shouldCleanupOrphansWithoutInterferingWithNormalActions() throws Exception {
+        // This test takes advantage of the immutable nature of contents with the in-place update branch
+        // disabled. If in-place updates are ever reenabled, we'll need a way to generate large numbers
+        // of orphaned contents for this test.
+        OwnerDTO owner2 = adminClient.owners().createOwner(Owners.random());
+        String contentIdPrefix = "content-id-";
+        int offset = 1000;
+        int numberOfContent = 100;
+        for (int i = 0; i < 10; i++) {
+            // Create a bunch of dummy contents
+            final int batchOffset = offset;
+            Set<String> owner1CreatedContentUuids = new HashSet<>();
+            Set<String> owner2CreatedContentUuids = new HashSet<>();
+            for (int j = batchOffset; j < batchOffset + numberOfContent; j++) {
+                String id = contentIdPrefix + j;
+                adminClient.ownerContent().createContent(owner.getKey(), Content.random().id(id));
+            }
+
+            // Attempt to update and create new contents to get into some funky race conditions with
+            // convergence and orphanization
+            Runnable owner1Updates = () -> {
+                for (int j = batchOffset; j < batchOffset + numberOfContent; j++) {
+                    try {
+                        String id = contentIdPrefix + j;
+                        ContentDTO createdContent = adminClient.ownerContent()
+                            .updateContent(owner.getKey(), id, Content.random().id(id));
+                        owner1CreatedContentUuids.add(createdContent.getUuid());
+                    }
+                    catch (Exception e) {
+                        // Swollow errors
+                    }
+                }
+            };
+            Thread owner1UpdatesThread = new Thread(owner1Updates);
+            owner1UpdatesThread.start();
+
+            Runnable owner2ContentGeneration = () -> {
+                for (int j = batchOffset; j < batchOffset + numberOfContent; j++) {
+                    String id = contentIdPrefix + j;
+                    ContentDTO createdContent = adminClient.ownerContent()
+                        .createContent(owner2.getKey(), Content.random().id(id));
+                    owner2CreatedContentUuids.add(createdContent.getUuid());
+                }
+            };
+            Thread owner2GenerationThread = new Thread(owner2ContentGeneration);
+            owner2GenerationThread.start();
+
+            // Wait for the threads to finish
+            owner1UpdatesThread.join(10000);
+            owner2GenerationThread.join(10000);
+
+            AsyncJobStatusDTO job = adminClient.jobs().scheduleJob("OrphanCleanupJob");
```

Comment:
I've commented on this above, but to make sure it's individually captured: this needs to occur between the worker threads starting and waiting for them to terminate.

Reply:
Updated